File: 9월 6일 - 맵과 셋/11723.cpp
// 맵과 셋: 11723 집합
#include <iostream>
#include <set>
using namespace std;

int main(){
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    set<int> s;
    int m, input;
    string cmd;
    cin >> m;
    while (m--){
        cin >> cmd;
        if (cmd == "add"){
            cin >> input;
            s.insert(input);
        } else if (cmd == "remove"){
            cin >> input;
            s.erase(input);
        } else if (cmd == "check"){
            cin >> input;
            if (s.find(input) == s.end()){ // input이 존재하지 않으면 true
                cout << "0\n";
            } else { 
                cout << "1\n"; 
            }
        } else if (cmd == "toggle"){
            cin >> input;
            if (s.find(input) == s.end()){
                s.insert(input);
            } else { 
                s.erase(input); 
            }
        } else if (cmd == "all"){
            for(int i=1; i<=20; i++){
                s.insert(i);
            }
        } else if (cmd == "empty"){
            s.clear();
        }
    }
    return 0;
}

File: 9월 6일 - 맵과 셋/Python/11723.py
import sys
input = sys.stdin.readline
M = int(input())
s = set()
for i in range(M):
	cmd = input().split()
	if len(cmd) == 1:
		if (cmd[0] == "all"):
			s = set([i for i in range(1, 21)])
		elif (cmd[0] == "empty"):
			s = set()
	else:
		cmd[1] = int(cmd[1])
		if (cmd[0] == "add"):
			s.add(cmd[1])
		elif (cmd[0] == "remove"):
			s.discard(cmd[1])
		elif (cmd[0] == "check"):
			print(1 if cmd[1] in s else 0)
		elif (cmd[0] == "toggle"):
			if (cmd[1] in s):
				s.discard(cmd[1])
			else:
				s.add(cmd[1])